.github/workflows/type-check.yml

```diff
@@ -17,13 +17,23 @@ jobs:
         env:
           RUNNER_CONTEXT: ${{ toJson(runner) }}
         run: echo "$RUNNER_CONTEXT"
+
       - name: Checkout
         uses: actions/checkout@v3
         with:
           fetch-depth: 1
 
       - name: Install Pnpm
-        run: corepack enable
+        uses: pnpm/action-setup@v2
```

Comment:
So why can't we use corepack?

Reply:
> We should specify the pnpm version to prevent pnpm v9 breaks the CI.

Don't worry, it will use version defined at `packageManager` : https://github.com/pnpm/action-setup#version

> So why can't we use corepack?

Occasionally, corepack could become unavailable if it's not installed globally. The pnpm action primarily depends on Node.js.

Reply:
get👍🏻